### ISSUE TITLE
Unexpected result for prefix scan with fixed prefix extractor

### ISSUE THREAD

**Issue by yunjing** [facebook/rocksdb#1617]
We use prefix scan with fixed prefix extractor. Our code follows the example in [wiki](https://github.com/facebook/rocksdb/wiki/Prefix-Seek-API-Changes):
```
auto iter = DB::NewIterator(ReadOptions());
for (iter.Seek(prefix); iter.Valid() && iter.key().starts_with(prefix); iter.Next()) {
       // do something
}
```
However, we are getting unexpected results when scanning merge entries. Specifically, we never call merge before putting the original value in the database, but our merge operator sometimes picks up existing values with `nullptr`. After reading the source code, we found that this is a somewhat known issue with a [fix](https://github.com/facebook/rocksdb/commit/9d50afc3b99eb04e7231c757e4aca8d79759b65a) added to the `ReadOptions`. We can get correct results using the following pattern for prefix scan:
```
ReadOptions options;
options.prefix_same_as_start = true;
auto iter = DB::NewIterator(options);
for (iter.Seek(prefix); iter.Valid(); iter.Next()) {
       // do something
}
```
It would be nice to fix the example in wiki.

**Comment by yunjing** [comment]
cc @lightmark @igorcanadi

**Comment by lightmark** [comment]
Hi @yunjing . I am curious about the `nullptr` you get from merge operator. Could you give an example about this?

**Comment by yunjing** [comment]
Not sure what you are looking for. Basically, in theory, `existing_value` in a merge operator can be `nullptr`. However, in the system I was referring to, we always call `Put` first before `Merge`. Therefore, we don't expect to see `nullptr` value in the merge operator. In practice, we did see it while running the prefix scan code path, i.e., when doing prefix scanning, we saw merge operator being called with `nullptr` existing value, if we use the code pattern found in wiki. The problem would go away if we use the second pattern by setting `prefix_same_as_start = true`
